sklearn/tree/_criterion.pyx

```diff
@@ -293,8 +292,9 @@ cdef class ClassificationCriterion(Criterion):
             i = samples[p]
 
             # w is originally set to be 1.0, meaning that if no sample weights
-            # are given, the default weight of each sample is 1.0
-            if sample_weight != NULL:
+            # are given, the default weight of each sample is 1.0.
+            # see: https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#syntax
```

Comment:
I think I would remove this comment because the code is clear and because the syntax has nothing much particular in this case.
```suggestion
```